components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -734,4 +807,78 @@ private static void clearAppCacheByInboundKey(ServiceProvider serviceProvider, S
             }
         }
     }
+
+    /**
+     * Clear the trusted app cache for the available platform types of all tenants depending on the
+     * availability of platform specific data.
+     *
+     * @param spTrustedAppMetadata Trusted app metadata of the service provider.
+     */
+    private static void clearTrustedAppCache(SpTrustedAppMetadata spTrustedAppMetadata) {
+
+        if (spTrustedAppMetadata != null) {
+            if (StringUtils.isNotEmpty(spTrustedAppMetadata.getAppleAppId())) {
+                clearIOSTrustedAppCache();
+            }
+            if (StringUtils.isNotEmpty(spTrustedAppMetadata.getAndroidPackageName())) {
+                clearAndroidTrustedAppCache();
+            }
+        }
+    }
+
+    /**
+     * Clear the trusted app cache for the available platform types of all tenants depending on the update of platform
+     * specific data, by  comparing the trusted app metadata objects of stored application and updated application.
+     *
+     * @param storedTrustedAppMetadata Trusted app metadata of the stored service provider.
+     * @param updatedTrustedAppMetadata Updated trusted app metadata of the service provider.
+     */
+    private static void clearTrustedAppCache(SpTrustedAppMetadata storedTrustedAppMetadata,
```

Comment:
Since we are not simply clearing the app cache but rather performing a validation and clearing the cache based on that lets rename the method definition to reflect that. Maybe something like ValidateAndClearTrustedAppCache